pkg/task/task.go

```diff
@@ -0,0 +1,72 @@
+package task
+
+import "time"
+
+// TaskState (kind: int) represents the last known state of a task.
+// A task can be in one of three states
+// StateScheduled: this is the initial state of the task when it enters the queue.
+// StateProcessing: once work begins on the task, it is put into this state.
+// StateComplete: work is no longer being done on this task. client should check task result.
+type TaskState int
+
+const (
+	StateScheduled TaskState = iota
+	StateProcessing
+	StateComplete
+)
+
+func (t TaskState) String() string {
+	return [...]string{
+		"StateRequested",
+		"StateProcessing",
+		"StateComplete",
+	}[t]
+}
+
+// ResultStatus (kind: int) is a status code for completed tasks.
+// ResultNone: initial status, No status, probably becasue the task is incomplete.
+// ResultSuccess: the task has completed without an error.
+// ResultPlanFail: the task has completed with a failure.
+// ResultTestgroundFail: testground encountered an error and the task has not been scheduled.
+type TaskResultStatus int
+
+const (
+	ResultNone TaskResultStatus = iota
+	ResultSuccess
+	ResultPlanFail
+	ResultTestgroundFail
+)
+
+func (t TaskResultStatus) String() string {
+	return [...]string{
+		"ResultNone",
+		"ResultSuccess",
+		"ResultPlanFail",
+		"ResultTestgroundFail",
+	}[t]
+}
+
+// DatedTaskState (kind: struct) is a TaskState with a timestamp.
+type DatedTaskState struct {
+	TaskState TaskState `json:"state"`
+	Entered   time.Time `json:"entered"`
```

Comment:
Nitpick, but I think `Created` is more idiomatic than `Entered`.
```suggestion
	Created   time.Time `json:"created"`
```